### PR TITLE
Avoid ClassCastException when recording telemetry on the privacy settings screen.

### PR DESCRIPTION
**Pull request by ionutbedregeanu**
For #6094
onSharedPreferenceChanges is also called when the value of Block cookies
is updated, which is a String value and can't be casted to Boolean

### CODE REVIEW

**Review by mcarare — APPROVED** [review]

File: app/src/main/java/org/mozilla/focus/settings/privacy/PrivacySecuritySettingsFragment.kt [hunk 1]
```diff
@@ -78,8 +78,11 @@ class PrivacySecuritySettingsFragment :
     }
 
     override fun onSharedPreferenceChanged(sharedPreferences: SharedPreferences, key: String) {
-        recordTelemetry(key, sharedPreferences.all[key] as Boolean)
-        updateStealthToggleAvailability()
+        val sharedPrefValue = sharedPreferences.all[key]
+        if (sharedPrefValue is Boolean) {
+            recordTelemetry(key, sharedPrefValue)
+            updateStealthToggleAvailability()
+        }
```
[the remark this answers is not in the document]
Comment (mcarare) [reply app/src/main/java/org/mozilla/focus/settings/privacy/PrivacySecuritySettingsFragment.kt]: @Amejia481 We did not send telemetry for that setting. If we want to, we could do it in `recordTelemetry` method.

[the remark this answers is not in the document]
Comment (ionutbedregeanu) [reply app/src/main/java/org/mozilla/focus/settings/privacy/PrivacySecuritySettingsFragment.kt]: I made some updates in recordTelemetry in order to be easier updated later. At this moment recordTelemetry records only for Send usage data, Stealth, Block potentially dangerous and deceptive sites, Use fingerprint to unlock app

[the remark this answers is not in the document]
Comment (mcarare) [reply app/src/main/java/org/mozilla/focus/settings/privacy/PrivacySecuritySettingsFragment.kt]: @Amejia481 I opened an [issue](https://github.com/mozilla-mobile/focus-android/issues/6097) for product/ DS feedback on the subject of telemetry for this setting.

[the remark this answers is not in the document]
Comment (Amejia481) [reply app/src/main/java/org/mozilla/focus/settings/privacy/PrivacySecuritySettingsFragment.kt]: Thanks for the clarification :)

**Review by Amejia481 — APPROVED** [review]

**Comment by mcarare** [comment]
removing `needs:landing` label until the commit message is updated.

**Comment by Amejia481** [comment]
👍🏽

**Pull request merged**